dash/include/dash/Matrix.h

```diff
@@ -95,23 +96,31 @@ template <
   typename T,
   dim_t NumDimensions,
   typename IndexT,
-  class PatternT >
+  class PatternT,
+  typename LocalMemSpaceT>
 class Matrix;
 /// Forward-declaration
 template <
   typename T,
   dim_t NumDimensions,
   dim_t CUR,
-  class PatternT >
+  class PatternT,
+  typename LocalMemSpaceT>
 class MatrixRef;
 /// Forward-declaration
 template <
   typename T,
   dim_t NumDimensions,
   dim_t CUR,
-  class PatternT >
+  class Pattern,
+  typename LocalMemSpaceT>
 class LocalMatrixRef;
 
+namespace halo {
```

Comment:
Why is this added to the `Matrix` class?

Reply:
Because we can then forward declare `HaloMatrixWrapper` and make it a friend. Otherwise c++ does not allow that.